### PR TITLE
Basic support of splitted ZIM files

### PR DESCRIPTION
**Pull request by mossroy**

### CODE REVIEW

**Comment by peter-x** [comment]
Please test
 - file discovery (was not sure how to test it)
 - concatenation of slices (had a weird bug there)

File: www/js/lib/zimArchive.js [hunk 1]
```diff
@@ -50,75 +50,62 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
...
+            }
         }
     };
 
     /**
+     * Searches the directory for all parts of a splitted archive.
+     * @param {Storage} storage storage interface
+     * @param {String} prefixPath path to the splitted files, missing the "aa" / "ab" / ... suffix.
+     * @returns {Promise} that resolves to the array of file objects found.
+     */
```
Comment (mossroy) [inline www/js/lib/zimArchive.js]: Cool : this is a much cleaner and concise way to do it.
It appears to work well on my device.
I'll just remove the unused variable and debug log

File: www/js/lib/zimfile.js [hunk 2]
```diff
@@ -81,16 +81,36 @@ define(['xzdec_wrapper', 'util', 'utf8'], function(xz, util, utf8) {
      */
     ZIMFile.prototype._readSlice = function(offset, size)
     {
-        var fileNumber = 0;
-        var offsetInActualFile = offset;
-        var fileSize = this._files[0].size;
-        if (this._files.length > 1 && offset > fileSize) {
-            // We need to find in which actual file the data must be read
-            fileNumber = Math.floor(offset / fileSize);
-            offsetInActualFile = offset - fileNumber * fileSize;
```
Comment (mossroy) [inline www/js/lib/zimfile.js]: This modification introduces a big performance regression (on device and on desktop too).
As we'd like to release a beta version very soon, I'll revert it for now (to be able to merge this pull request), and create a separate issue for that.

Comment (peter-x) [inline www/js/lib/zimfile.js]: That's weird. Are you saying the regression also exists for single-file archives?

**Pull request merged**